packages/xrpl-types/src/types.rs

```diff
@@ -802,7 +802,12 @@ impl XRPLCurrency {
     }
 
     fn from_nonstandard_str(s: &str) -> Self {
-        XRPLCurrency(s.as_bytes().try_into().expect("should be 20 bytes"))
+        XRPLCurrency(
+            hex::decode(s)
+                .expect("invalid nonstandard currency hex")
+                .try_into()
+                .expect("should be 20 bytes"),
+        )
```

Comment:
can this be triggered by user input at all? I'm wondering if it should return an error